src/services/rfqm_service.ts

```diff
@@ -0,0 +1,95 @@
+import { MarketOperation, QuoteRequestor } from '@0x/asset-swapper';
+import { BigNumber } from '@0x/utils';
+
+import { RFQT_REQUEST_MAX_RESPONSE_MS } from '../config';
+import { NULL_ADDRESS } from '../constants';
+
+export interface FetchIndicativeQuoteParams {
+    apiKey: string;
+    amount: BigNumber; // in terms of the amount the taker is selling
+    buyToken: string;
+    sellToken: string;
+    takerAddress?: string;
+}
+
+export interface FetchIndicativeQuoteResponse {
+    buyAmount: BigNumber;
+    buyTokenAddress: string;
+    price: BigNumber;
+    sellAmount: BigNumber;
+    sellTokenAddress: string;
+}
+
+/**
+ * RfqmService is the coordination layer for HTTP based RFQM flows
+ */
+export class RfqmService {
+    constructor(private readonly _quoteRequestor: QuoteRequestor) {}
+
+    /**
+     * Fetch the best indicative quote available.
+     */
+    public async fetchIndicativeQuoteAsync(params: FetchIndicativeQuoteParams): Promise<FetchIndicativeQuoteResponse> {
+        const { amount, sellToken, buyToken, apiKey } = params;
+
+        // Map params to the terminology of Quote Requestor
+        const marketOperation = MarketOperation.Sell;
+        const assetFillAmount = amount;
+        const makerToken = buyToken;
+        const takerToken = sellToken;
+
+        // Fetch quotes
+        const opts = {
+            takerAddress: NULL_ADDRESS,
+            txOrigin: NULL_ADDRESS, // TODO - set to worker registry
+            apiKey,
+            intentOnFilling: false, // TODO - safe to hardcode?
+            isIndicative: true,
+            makerEndpointMaxResponseTimeMs: RFQT_REQUEST_MAX_RESPONSE_MS,
+            nativeExclusivelyRFQ: true,
+            altRfqAssetOfferings: {}, // TODO - set this to the alt rfq asset offerings
+            isLastLook: true,
+        };
+        const indicativeQuotes = await this._quoteRequestor.requestRfqmIndicativeQuotesAsync(
+            makerToken,
+            takerToken,
+            assetFillAmount,
+            marketOperation,
+            undefined,
+            opts,
+        );
+
+        // Filter out quotes that:
+        // - are for the wrong pair
+        // - cannot fill 100 % of the requested amount
+        //
+        // And sort by best price
+        const sortedQuotes = indicativeQuotes
```

Comment:
This is awersome! is there any chance we could move this to a separte function, so that we can properly unit test it?

Reply:
I've been thinking about how to pull out this logic, but want to wait, because this logic will be shared with firm quotes, but the types are not the same.

All said, the current unit test coverage does exercise this code! Perhaps not exhaustively, but I believe the major cases are there

Will revisit plucking out this code when I implement the firm quote endpoint